app/operations/memberships/create.rb

```diff
@@ -12,7 +12,9 @@ def execute
       raise Unauthorized unless user_group.verify_join_token(join_token)
       raise Unauthorized unless user.id == api_user.id
 
-      Membership.create! user: api_user.user, user_group: user_group, state: :active
+      membership = Membership.find_or_initialize_by(user: api_user.user, user_group: user_group)
```

Comment:
should we explicitly query for deactivated memberships here to match the spec? 
```
find_or_initialize_by(
  user: api_user.user, 
  user_group: user_group, 
  state: :inactive
)
```

Reply:
I want to get an any existing membership regardless of state with that line, then just make sure it's active with the next one.